engines/alg/detection.cpp

```diff
@@ -0,0 +1,69 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "base/plugins.h"
+#include "engines/advancedDetector.h"
+
+#include "alg/alg.h"
+#include "alg/detection.h"
+#include "alg/detection_tables.h"
+
+static const PlainGameDescriptor algGames[] = {
+	{ "cpatrol", "Crime Patrol" },
+	{ "dwars", "Drug Wars" },
+	{ "johnroc", "Who Shot Johnny Rock?" },
+	{ "lbhunter", "The Last Bounty Hunter" },
+	{ "maddog", "Mad Dog McCree" },
+	{ "maddog2", "Mad Dog II: The Lost Gold" },
+	{ "spirates", "Space Pirates" },
+	{ nullptr, nullptr }
```

Comment:
What about the following?
- Gallagher's Gallery
- Fast Draw Showdown
- Shootout at Old Tucson

Also, it would be great if these are sorted by year (with Mad Dog McCree being the oldest, in 1990, and Shootout at Old Tucson the newest, in 1994)

Reference: https://en.wikipedia.org/wiki/American_Laser_Games